src/test/descriptor_tests.cpp

```diff
@@ -161,7 +161,7 @@ void DoCheck(std::string prv, std::string pub, const std::string& norm_pub, int
     // We must be able to estimate the max satisfaction size for any solvable descriptor top descriptor (but combo).
     const bool is_nontop_or_nonsolvable{!parse_priv->IsSolvable() || !parse_priv->GetOutputType()};
     const auto max_sat_maxsig{parse_priv->MaxSatisfactionWeight(true)};
-    const auto max_sat_nonmaxsig{parse_priv->MaxSatisfactionWeight(true)};
+    const auto max_sat_nonmaxsig{parse_priv->MaxSatisfactionWeight(false)};
```

Comment:
Haven't verified all the scenarios but maybe you could verify that `max_sat_nonmaxsig < max_sat_maxsig` (or <= if that causes any complication).

Reply:
Done.